components/asset.rst

```diff
@@ -0,0 +1,255 @@
+
+Request Context Aware Assets
+............................
+
+If you are also using the HttpFoundation component in your project, for example
+in a Symfony application, the :class:`Symfony\Component\Asset\PathPackage` class
+can take into account the context of the current request::
+
+    use Symfony\Component\Asset\PathPackage;
+    use Symfony\Component\Asset\Context\RequestStackContext;
+    // ...
+
+    $package = new PathPackage('/static/images', new StaticVersionStrategy('v1'));
+    $package->setContext(new RequestStackContext($requestStack));
+
+    echo $package->getUrl('/logo.png');
+    // result: /somewhere/static/images/logo.png?v1
+
+When the request context is set, in addition to the configured base path,
+:class:`Symfony\Component\Asset\PathPackage` also prepends the current request
+base URL (``/somewhere/`` in this example) to assets. This allows your website
+to be hosted anywhere under the web server root directory.
+
+Absolute Assets and CDNs
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Applications that host their assets on different domains and CDNs (*Content
+Delivery Networks*) should use the :class:`Symfony\Component\Asset\UrlPackage`
+class to generate absolute URLs for their assets::
+
+    use Symfony\Component\Asset\UrlPackage;
+    // ...
+
+    $package = new UrlPackage(
+        'http://static.example.com/images/',
+        new StaticVersionStrategy('v1')
+    );
+
+    echo $package->getUrl('/logo.png');
+    // result: http://static.example.com/images/logo.png?v1
+
+In case you serve assets from more than one domain to improve application
+performance, pass an array of URLs as the first argument of
+:class:`Symfony\Component\Asset\UrlPackage` constructor::
+
+    use Symfony\Component\Asset\UrlPackage;
+    // ...
+
+    $urls = array(
+        'http://static1.example.com/images/',
+        'http://static2.example.com/images/',
+    );
+    $package = new UrlPackage($urls, new StaticVersionStrategy('v1'));
+
+    echo $package->getUrl('/logo.png');
+    // result: http://static1.example.com/images/logo.png?v1
+
+The selection of the domain which will serve the asset is deterministic, meaning
+that each asset will be always served by the same domain. This behavior simplifies
+the management of HTTP cache.
+
+Request Context Aware Assets
+............................
+
+Similarly to application-relative assets, absolute assets can also take into
+account the context of the current request. In this case, only the request
+scheme is considered, in order to select the appropriate base URL (HTTPs or
+protocol-relative URLs for HTTPs requests, any base URL for HTTP requests)::
+
+    use Symfony\Component\Asset\UrlPackage;
+    use Symfony\Component\Asset\Context\RequestStackContext;
+    // ...
+
+    $package = new UrlPackage(
+        array('http://example.com/', 'https://example.com/'),
+        new StaticVersionStrategy('v1')
+    );
+    $package->setContext(new RequestStackContext($requestStack));
```

Comment:
this is wrong. there is no ``setContext`` method anymore (it was removed during the review of the component, making the package immutable). The context is actually the last constructor argument.